clang/lib/Sema/SemaChecking.cpp

```diff
@@ -851,6 +851,50 @@ class ScanfDiagnosticFormatHandler
   }
 };
 
+/// `I` points to the next character of `%p` format.
+/// This functon checks if the subsequent character can be linux kernel's
+/// extnded format specifier
+static inline constexpr bool canBeLinuxFormatExtension(const char *I,
+                                                       const char *E) {
+  assert(I < E && "format string not yet exhausted");
+  // Kernel Document: https://docs.kernel.org/core-api/printk-formats.html
+  switch (*I) {
```

Comment:
The URL with anchor https://docs.kernel.org/core-api/printk-formats.html#pointer-types may be more precise to use in the comment.

Reply:
I think it would also be helpful to say something along the lines of:

see the `pointer()` function in https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/lib/vsprintf.c.